File: src/api/java/com/ultreon/mods/advanceddebug/api/client/menu/IDebugRenderContext.java
package com.ultreon.mods.advanceddebug.api.client.menu;

import net.minecraft.network.chat.Component;

public interface IDebugRenderContext {
    void left(Component text, Object object, Object... objects);

    void left(String text, Object object, Object... objects);

    void left(Component text);

    void left(String text);

    void left();

    void right(Component text, Object object, Object... objects);

    void right(String text, Object object, Object... objects);

    void right(Component text);

    void right(String text);

    void right();

    void top(Component text);

    void top(String text);

    void top();
}

File: common/src/main/kotlin/com/ultreon/mods/advanceddebug/api/client/menu/IDebugRenderContext.kt
package com.ultreon.mods.advanceddebug.api.client.menu

import net.minecraft.network.chat.Component

interface IDebugRenderContext {
    fun left(text: Component, `object`: Any?, vararg objects: Any?)
    fun left(text: String, `object`: Any?, vararg objects: Any?)
    fun left(text: Component)
    fun left(text: String)
    fun left()
    fun right(text: Component, `object`: Any?, vararg objects: Any?)
    fun right(text: String, `object`: Any?, vararg objects: Any?)
    fun right(text: Component)
    fun right(text: String)
    fun right()
    fun top(text: Component)
    fun top(text: String)
    fun top()
}